### PR TITLE
Feature/operator bit not

### PR DESCRIPTION
**Pull request by gurugio**
In the last PR, I pushed a implementation for bot of `BIT_NOT` function and ~ operator.
As review comment, I removed implementation of `BIT_NOT` function and pushed only ~ operator in this PR.
Please review.

I will add more test cases if the coverage test result is not good enough.

### CODE REVIEW

**Comment by gurugio** [comment]
I am going to add some more tests to increase coverage.

File: core/src/executor/evaluate/evaluated.rs [hunk 1]
```diff
@@ -252,6 +252,17 @@ impl<'a> Evaluated<'a> {
         .map(Evaluated::from)
     }
 
+    pub fn unary_bit_not(&self) -> Result<Evaluated<'a>> {
+        match self {
+            Evaluated::Literal(v) => Value::try_from(v).and_then(|v| v.unary_bit_not()),
+            Evaluated::Value(v) => v.unary_bit_not(),
+            Evaluated::StrSlice { source, range } => {
+                Err(EvaluateError::UnsupportedUnaryBitNot(source[range.clone()].to_owned()).into())
+            }
```
Comment (ever0de) [inline core/src/executor/evaluate/evaluated.rs:261]: Can we also add test code for this case?
https://coveralls.io/builds/61510412/source?filename=core%2Fsrc%2Fexecutor%2Fevaluate%2Fevaluated.rs#L259

Comment (gurugio) [reply core/src/executor/evaluate/evaluated.rs:261]: Let me try.
Thank you for the review.

Comment (gurugio) [reply core/src/executor/evaluate/evaluated.rs:313]: I pushed a new patch to test EvaluateError::UnsupportedUnaryBitNot error.
I am not sure why it cannot tirgger coverage test.

**Comment by gurugio** [comment]
I will add a test case to cover Null data.

File: core/src/ast/operator.rs [hunk 2]
```diff
@@ -18,6 +19,7 @@ impl ToSql for UnaryOperator {
             UnaryOperator::Minus => "-".to_owned(),
             UnaryOperator::Not => "NOT ".to_owned(),
             UnaryOperator::Factorial => "!".to_owned(),
+            &UnaryOperator::BitNot => "~".to_owned(),
```
Comment (panarch) [inline core/src/ast/operator.rs:22]: 1. `&` is unnecessary. we can remove this.
2. could we use `Bitwise` prefix rather than `Bit` because we already use `bitwise` for other operators. e.g. `BinaryOperator::BitwiseAnd` and `BinaryOperator::BitwiseShiftLeft`

Comment (gurugio) [reply core/src/ast/operator.rs:22]: 1. Ok. It's my mistake.
2. Ok. I will change it to BitwiseNot.

**Comment by gurugio** [comment]
Hi,
I added test cases covering all new codes.
And change names as panarch's review.

Now I think it's ready to be merged.

File: core/src/executor/evaluate/expr.rs [hunk 3]
```diff
@@ -66,6 +66,7 @@ pub fn unary_op<'a>(op: &UnaryOperator, v: Evaluated<'a>) -> Result<Evaluated<'a
         UnaryOperator::Minus => v.unary_minus(),
         UnaryOperator::Not => v.try_into().map(|v: bool| Evaluated::from(Value::Bool(!v))),
         UnaryOperator::Factorial => v.unary_factorial(),
+        &UnaryOperator::BitwiseNot => v.unary_bitwise_not(),
```
Comment (panarch) [inline core/src/executor/evaluate/expr.rs:69]: we can remove this `&`, too.

File: core/src/translate/operator.rs [hunk 4]
```diff
@@ -13,6 +13,7 @@ pub fn translate_unary_operator(sql_unary_operator: &SqlUnaryOperator) -> Result
         SqlUnaryOperator::Minus => Ok(UnaryOperator::Minus),
         SqlUnaryOperator::Not => Ok(UnaryOperator::Not),
         SqlUnaryOperator::PGPostfixFactorial => Ok(UnaryOperator::Factorial),
+        &SqlUnaryOperator::PGBitwiseNot => Ok(UnaryOperator::BitwiseNot),
```
Comment (panarch) [inline core/src/translate/operator.rs:16]: here, too.

**Comment by gurugio** [comment]
Hi,
I just pushed PR again
- remove unnecessary &
- use test! macro for test suite

**Review by panarch — APPROVED** [review]
Looks all nice! Thanks a lot for the contribution 👍 👍
